Src/HALALMock/Services/Communication/FDCAN/FDCAN.cpp

```diff
@@ -89,43 +92,52 @@ bool FDCAN::transmit(uint8_t id, uint32_t message_id, const char* data, FDCAN::D
 		return false;
 	}
 
-	instance->tx_header.Identifier = message_id;
+    size_t buffer_len = sizeof(message_id)+ sizeof(dlc) + dlc;
```

Comment:
dlc is an enum and the variant values problably aren't 4 for BYTES4, 8 for BYTES8, etc. Create a new function `ssize_t dlc_to_number_of_bytes(FDCAN::DLC dlc)` which returns this number and inside do a switch:
```c++
switch (dlc) {
    case DLC::BYTES4:
        return 4;
    case DLC::BYTES8:
        return 8;
    // rest of cases
```

Reply:
this is a good point